src/node/vscode.ts

```diff
@@ -51,14 +51,20 @@ export class VscodeProvider {
 
     logger.debug("setting up vs code...")
     return new Promise<ipc.WorkbenchOptions>((resolve, reject) => {
+      const onExit = (code: number | null) => reject(new Error(`VS Code exited unexpectedly with code ${code}`))
+
       vscode.once("message", (message: ipc.VscodeMessage) => {
         logger.debug("got message from vs code", field("message", message))
+        vscode.off("error", reject)
```

Comment:
Why do we want to unbind if an error occurs with the vscode process? Wouldn't we still want to be notified?

Reply:
Here all we do is reject if an error occurs, so once the promise has fulfilled keeping this handler will call `reject` but have no effect so we might as well unbind it.

But we do bind a different handler to `error` during spawn and log the error there as well as unset the process so we can spawn another one.

Reply:
I guess I'm confused why we're binding twice? What's wrong with the earlier bind? I understand rejecting the promise but why duplicate `onExit`?

Reply:
By duplicate `exit` binds do you mean the one here in `initialize` and the bind in `fork`?

Reply:
Yes!

Reply:
Ah I see, you wait for the `"ready"` message and then `"options"` message. I think we can definitely structure this code to reduce the duplication. Feel free to defer though for a later PR.

Reply:
Mmm yeah `fork` happens once when spawning VS Code and `initialize` happens every time we load the page to send and receive initialization options. Def agree that we could create some kind of generalized wrapper around a process that handles all the binds for you whenever you want to spawn or communicate with it since there's quite a bit of overlap. For now I'll defer to later :smile:

Reply:
Ahh yeah I re-read your comment and realized it's not just "quite a bit of overlap" it's literally the same code lol just a different message.

Reply:
Gonna just rewrite it now actually.